pkg/apps/application-list.jsx

```diff
@@ -117,14 +120,36 @@ export const ApplicationList = ({ metainfo_db, appProgress, appProgressTitle, ac
         }
     }
 
+    async function check_missing_data(packages) {
+        // avoid a metadata refresh at page load, too expensive; trust our OS map
+        try {
+            const data = await check_missing_packages(packages, null, true);
```

Comment:
`check_missing_packages` feels like a too big hammer for this. We don't need the refreshing, simulating, and details getting that it does. I think we should just make a dedicated function in `pkg/apps/packagekit.js` that just runs "Resolve". There are already enough helper functions there that should make this pretty straightforward.

Reply:
Good point, done.